sympy/core/tests/test_arit.py

```diff
@@ -2321,3 +2323,16 @@ def test_issue_18507():
 def test_issue_17130():
     e = Add(b, -b, I, -I, evaluate=False)
     assert e.is_zero is None # ideally this would be True
+
+
+def test_issue_21034():
+    x = Symbol('x', real=True, nonzero=True)
+    f1 = x*(-x**4/asin(5)**4 - x*sinh(x + log(asin(5))) + 5)
+    e = -I*log((re(asin(5)) + I*im(asin(5)))/sqrt(re(asin(5))**2 + im(asin(5))**2))
+    e2 = -I*log((re(asin(5)) + I*im(asin(5)))/sqrt(re(asin(5))**2 + im(asin(5))**2))/pi
+    assert integrate(f1, x) == \
+        -x**6/(6*asin(5)**4) - x**2*cosh(x + log(asin(5))) + 5*x**2/2 + 2*x*sinh(x + log(asin(5))) - 2*cosh(x + log(asin(5)))
+    assert f1.is_zero is None
```

Comment:
Looking at this again there's really just one bug here which is this:
```
expr = -I*log((re(asin(5)) + I*im(asin(5)))/sqrt(re(asin(5))**2 + im(asin(5))**2))/pi
expr.round() # hangs
```
We can just add a test for that here in this file. The integral test should go in `sympy.integrals` somewhere.

Reply:
on it. 
also in `sympy.integrals` I'm adding it to `sympy/interals/test_integrals.py` is it fine?